keps/0019-package-api-versioning.md

```diff
@@ -2,64 +2,374 @@
 ---
 
 # Versioning of Operator Packages
 
-## Table of Contents
+**Table of Contents**
 
-* [Summary](#summary)
-* [Motivation](#motivation)
-  * [Goals](#goals)
-  * [Non-Goals](#non-goals)
-* [Proposal](#proposal)
-  * [Support multiple API versions in the package repository](#support-multiple-api-versions-in-the-package-repository)
-  * [Install a specific application version](#install-a-specific-application-version)
-  * [Naming of package tarballs](#naming-of-package-tarballs)
-  * [Semantic versioning](#semantic-versioning)
-  * [Risks and Mitigations](#risks-and-mitigations)
-* [Implementation History](#implementation-history)
+- [Concepts](#concepts)
+    - [KUDO version](#kudo-version)
+    - [KUDO API version](#kudo-api-version)
+    - [Operator](#operator)
+    - [Operator user (user)](#operator-user-user)
+    - [Operator developer (developer)](#operator-developer-developer)
+    - [Package](#package)
+    - [Package version](#package-version)
+    - [Package Registry](#package-registry)
+    - [Application](#application)
+    - [Application version (app version)](#application-version-app-version)
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+        - [Non-functional](#non-functional)
+        - [Functional](#functional)
+    - [Non-Goals](#non-goals)
+- [Current State](#current-state)
+- [Proposals](#proposals)
+    - [With regards to package version and operator version](#with-regards-to-package-version-and-operator-version)
+        - [Singular package/operator version (largely, the current state of things)](#singular-packageoperator-version-largely-the-current-state-of-things)
+        - [Composite package/operator version](#composite-packageoperator-version)
+    - [Support multiple KUDO API versions in the package repository](#support-multiple-kudo-api-versions-in-the-package-repository)
+    - [Install a specific application version](#install-a-specific-application-version)
+    - [Naming of package tarballs](#naming-of-package-tarballs)
+    - [Semantic versioning](#semantic-versioning)
+    - [Risks and Mitigations](#risks-and-mitigations)
+- [Open Questions](#open-questions)
+- [Implementation History](#implementation-history)
+
+## Concepts
+
+### KUDO version
+
+A released version of KUDO (CLI, controller-manager), e.g. 0.7.0, 0.8.0, etc.
+
+### KUDO API version
+
+The internal version for the KUDO implementation, e.g. v1alpha1, v1beta1, v1, etc.
+
+### Operator
+
+A [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) built on KUDO.
+
+### Operator user (user)
+
+Someone that installs or manages an _operator_.
+
+### Operator developer (developer)
+
+Someone that builds or maintains an _operator_.
+
+### Package
+
+An archive that contains all necessary files for running an _operator_.
+
+### Package version
+
+A single version identifier that represents all of the underlying exact versions or minimum version requirements for an _operator_:
```

Comment:
so this is the version I will use when doing `kudo install xxx -v myversion`?

Reply:
Correct. I'll add your clarification.